pkg/postage/postagecontract/contract.go

```diff
@@ -204,6 +242,42 @@ func (c *postageContract) CreateBatch(ctx context.Context, initialBalance *big.I
 	return nil, ErrBatchCreate
 }
 
+func (c *postageContract) TopUpBatch(ctx context.Context, batchID []byte, topUpAmount *big.Int) error {
+
+	batch, err := c.postageStorer.Get(batchID)
+	if err != nil {
+		return err
+	}
+
+	totalAmount := big.NewInt(0).Mul(topUpAmount, big.NewInt(int64(1<<batch.Depth)))
```

Comment:
since `topUpAmount` is never used below, you can just call `topUpAmount.Mul(big.NewInt(int64(1<<batch.Depth)))`

Reply:
Mul takes 2 arguments and puts result into the calling object.